### ISSUE TITLE
`no-undefined-types` - false positive on `@link` to symbol declared inside `declare global` block

### ISSUE THREAD

**Issue by DetachHead** [gajus/eslint-plugin-jsdoc#1701]
## Expected behavior
no error

## Actual behavior
```
  6:1  error  The type 'foo' is undefined  jsdoc/no-undefined-types

✖ 1 problem (1 error, 0 warnings)
```

the typescript language server is able to resolve `foo` since i can go to definition on it, so this error is a false positive

## ESLint Config

```js
/** @type {import('eslint').Linter.Config} */
const config = {
  extends: [
    'plugin:@typescript-eslint/recommended',
  ],
  parser: '@typescript-eslint/parser',
  parserOptions: {
    ecmaVersion: 12,
    sourceType: 'module',
    project: './tsconfig.json',
  },
  rules: {
    'import/no-undefined-types': 'error',
  },
};

module.exports = config;
```

## ESLint sample

```ts
declare global {
  const foo: number;
}

/**
 * {@link foo}
 */
export const bar = 1;
```

## Environment

- Node version: 22.17.0
- ESLint version 8.57.1
- `eslint-plugin-jsdoc` version: 63.0.2
